### PR TITLE
add batched evaluateVGLWithSpin to SpinorSet

### PR DESCRIPTION
**Pull request by camelto2**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes
This PR adds the mw_evaluateVGLWithSpin API for SpinorSet. Similar implementation as the mw_evaluate_notranspose. 

I also updated the unit test to check this function, as well as updated the unit test that used LCAO spinors since I overlooked that in the last PR. 

I think I probably need to implement a few other SpinorSet mw_ APIs, but still looking through the code to see exactly what is needed. I will add other SpinorSet mw_ calls as needed in future PRs. 

Part of #3712 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
macOSX

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCWaveFunctions/SPOSet.cpp [hunk 1]
```diff
@@ -120,7 +134,8 @@ void SPOSet::evaluate_notranspose_spin(const ParticleSet& P,
                                        ValueMatrix_t& d2logdet,
                                        ValueMatrix_t& dspinlogdet)
 {
-  APP_ABORT("Need specialization of " + className + "::evaluate_notranspose_spin(P,iat,psi,dpsi,d2logdet, dspin_logdet) (vector quantities)\n");
+  APP_ABORT("Need specialization of " + className +
```
Comment (ye-luo) [inline src/QMCWaveFunctions/SPOSet.cpp:137]: runtime_error

File: src/QMCWaveFunctions/tests/test_MO_spinor.cpp [hunk 2]
```diff
@@ -78,7 +79,7 @@ void test_lcao_spinor()
 
   Libxml2Document doc;
   bool okay = doc.parseFromString(particles);
-  REQUIRE(okay);
+  CHECK(okay);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/tests/test_MO_spinor.cpp:82]: I think it is better to have REQUIRE here. If parsing input goes wrong, all the rest will just produce garbage. So better to stop here.

File: src/QMCWaveFunctions/tests/test_MO_spinor.cpp [hunk 3]
```diff
@@ -92,7 +93,7 @@ void test_lcao_spinor()
     if (cname == "sposet")
       spo = bb.createSPOSet(element);
   });
-  REQUIRE(spo);
+  CHECK(spo);
```
Comment (ye-luo) [inline src/QMCWaveFunctions/tests/test_MO_spinor.cpp:96]: REQUIRE is better as a safeguard

File: src/QMCWaveFunctions/tests/lcao_spinor_test.py [hunk 4]
```diff
@@ -130,13 +130,13 @@ def val(self,pos):
         norm = self.norm()
```
Comment (ye-luo) [inline src/QMCWaveFunctions/tests/lcao_spinor_test.py:130]: I saw above an h5 file was removed. No more needed?

File: src/QMCWaveFunctions/SPOSet.cpp (hunk 1 above)
Comment (camelto2) [reply src/QMCWaveFunctions/SPOSet.cpp]: I went ahead and replaced all  the APP_ABORTS in the  file, not just this spin function

File: src/QMCWaveFunctions/tests/test_MO_spinor.cpp (hunk 2 above)
Comment (camelto2) [reply src/QMCWaveFunctions/tests/test_MO_spinor.cpp]: Agreed, I overlooked these when I replaced all the REQUIRES with CHECKS

File: src/QMCWaveFunctions/tests/test_MO_spinor.cpp (hunk 3 above)
Comment (camelto2) [reply src/QMCWaveFunctions/tests/test_MO_spinor.cpp]: done

File: src/QMCWaveFunctions/SPOSet.cpp (hunk 1 above)
Comment (ye-luo) [reply src/QMCWaveFunctions/SPOSet.cpp]: Appreciated

File: src/QMCWaveFunctions/tests/lcao_spinor_test.py (hunk 4 above)
Comment (camelto2) [reply src/QMCWaveFunctions/tests/lcao_spinor_test.py:130]: the file wasn't removed, it just got overwritten. The lcao_spinor_test.py writes the h5 file that is read by SPOSetBuilder in the test_MO_spinor.py. When I reran it to add new reference values, it just overwrote  that file

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Pull request merged**